gossipd/gossip.c

```diff
@@ -1765,12 +1765,62 @@ static void try_connect(struct reaching *reach)
 	io_new_conn(reach, fd, conn_init, reach);
 }
 
+static void try_connect_local_socket(struct daemon *daemon, const struct pubkey *id, const struct addrhint *a)
```

Comment:
Might conflict with ongoing changes in #1322, that is higher priority than this I think, so please monitor that PR when it gets merged in and see if this gets conflicted (maybe not a text-level conflict but some of the functions this `try_connect_local_socket` calls might change their signatures and exact behavior).  The first thing I can think of is that `reach->succeeded` has been removed in #1322, but please see that PR more for other changes that might affect this code.